bloom-housing/bloom
# 1153 Dropzone-style file upload component

# Pull Request Template

## Issue

Addresses #1153 

## Description

This PR provides a "dropzone" component that can be used for file uploads. It accepts an `uploader` prop which is a function which receives a [`file` object](https://developer.mozilla.org/en-US/docs/Web/API/File). You can upload the file by any means to any service. This PR includes one such implementation for Cloudinary. The component also accepts a `progress` prop, which when non-zero, will display a progress bar indicating upload progress, or will hide both dropzone and progress bar when 100.

Additional notes:
* What is done *after* a file gets uploaded is beyond the scope of the component. You could display an image thumbnail or a link to the file or whatever.
* The original issue requested multiple file upload support, but for expediency this PR only supports single file uploads.

## Type of change


## How Can This Be Tested/Reviewed?

There is a Dropzone story which provides two "Storybook knobs": a Cloudinary cloud name and upload preset. Once you supply a valid cloud name/upload preset, you can upload an image file, and once uploaded it will display a small preview of the image. (Note that the image preview is of course just part of the story, and not the component itself).


## Checklist:
